lib/core/utils/node-serializer.js

```diff
@@ -0,0 +1,133 @@
+import assert from './assert';
+import DqElement from './dq-element';
+
+let customSerializer = null;
+
+const nodeSerializer = {
+  /**
+   * @param {Object} newSerializer
+   * @property {Function} toSpec (Optional) Converts an Element or VirtualNode to a "spec", a form
```

Comment:
Based on how this function is called it only works with DqElements and not Elements or VirtualNodes. Also should this function be updated to be called `dqElmToSpec` as that's what it's doing now (and where it's called from)?

```suggestion
   * @property {Function} toSpec (Optional) Converts a DqElement to a "spec", a form
```